src/safeposix/syscalls/net_calls.rs

```diff
@@ -159,35 +180,49 @@ impl Cage {
 
     //creates a sockhandle if none exists, otherwise this is a no-op
     pub fn force_innersocket(sockhandle: &mut SocketHandle) {
+        //If innersocket is not available, then insert a socket into innersocket
         if let None = sockhandle.innersocket {
+            //Create a new socket
             let thissock =
                 interface::Socket::new(sockhandle.domain, sockhandle.socktype, sockhandle.protocol);
 
+            //Loop through socket options and check which ones are set
```

Comment:
This is a weird little bit of code.  I think you want to explain that because you need to make separate calls to set different socket options (i.e. you can't set multiple flags at once) you loop here.